inginious/agent/__init__.py

```diff
@@ -161,9 +162,9 @@ async def __handle_new_job(self, message: BackendNewJob):
         await ZMQUtils.send(self.__backend_socket, AgentJobStarted(message.job_id))
 
         try:
-            if message.environment not in self.environments:
-                self._logger.warning("Task %s/%s ask for an unknown environment %s (not in aliases)", message.course_id, message.task_id,
-                                     message.task_data['environment'])
+            if message.environment_type not in self.environments or not any(message.environment == x['name'] for x in self.environments[message.environment_type]):
```

Comment:
Wouldn't it be easier to have

    "type": {
                "name": {
                    "id": "env img id",   # 
                    "created": 12345678,  # create date
                    "ports": [22, 434],   # list of ports needed
                }
            }
and check for ``message.environment in self.environments[message.environment_type]`` ?